meta/data.go

```diff
@@ -177,6 +177,43 @@ func (data *Data) DropDatabase(name string) error {
 	return ErrDatabaseNotFound
 }
 
+// RenameDatabase renames a database.
+// Returns an error if oldName or newName is blank
+// or if a database with the newName already exists
+// or if a database with oldName does not exist
+func (data *Data) RenameDatabase(oldName, newName string) error {
+	if newName == "" || oldName == "" {
+		return ErrDatabaseNameRequired
+	}
+	if data.Database(newName) != nil {
+		return ErrDatabaseExists
+	}
+	if data.Database(oldName) == nil {
+		return ErrDatabaseNotFound
+	}
+	// find database named oldName and rename it to newName
+	for i := range data.Databases {
+		if data.Databases[i].Name == oldName {
+			data.Databases[i].Name = newName
+			data.switchDatabaseUserPrivileges(oldName, newName)
+			// TODO should rename the databases used in continuous queries
```

Comment:
@linearb -- I think they would fail at query time, which is quite difficult to track down in the current system. I think we should error for now, and take the conservative approach with this new command. When we improve CQ handling we could remove this constraint.

Thoughts?